src/web/components/Onwards/Onwards.tsx

```diff
@@ -1,28 +1,170 @@
 import React from 'react';
 
-import { useApi } from '@root/src/web/components/lib/api';
+import { useApi } from '@frontend/web/components/lib/api';
 
 import { OnwardsLayout } from './OnwardsLayout';
 
+// This list is a direct copy from https://github.com/guardian/frontend/blob/6da0b3d8bfd58e8e20f80fc738b070fb23ed154e/static/src/javascripts/projects/common/modules/onward/related.js#L27
+// If you change this list then you should also update ^
+// order matters here (first match wins)
+export const WHITELISTED_TAGS = [
+    // sport tags
+    'sport/cricket',
+    'sport/rugby-union',
+    'sport/rugbyleague',
+    'sport/formulaone',
+    'sport/tennis',
+    'sport/cycling',
+    'sport/motorsports',
+    'sport/golf',
+    'sport/horse-racing',
+    'sport/boxing',
+    'sport/us-sport',
+    'sport/australia-sport',
+
+    // football tags
+    'football/championsleague',
+    'football/premierleague',
+    'football/championship',
+    'football/europeanfootball',
+    'football/world-cup-2014',
+
+    // football team tags
+    'football/manchester-united',
+    'football/chelsea',
+    'football/arsenal',
+    'football/manchestercity',
+    'football/tottenham-hotspur',
+    'football/liverpool',
+];
+
+const joinUrl = (parts: string[]) => {
+    // Remove any leading or trailing slashes from all parts and then join cleanly on
+    // a single slash - prevents malformed urls
+    const trimmed = parts
+        .map(part => {
+            // Trim left
+            if (part.substr(0, 1) === '/') return part.slice(1);
+            return part;
+        })
+        .map(part => {
+            // Trim right
+            if (part.substr(part.length - 1, 1) === '/')
+                return part.slice(0, -1);
+            return part;
+        });
+
+    return trimmed.join('/');
+};
+
+const firstPopularTag = (pageTags: string[], isPaidContent: boolean) => {
+    // This function looks for the first tag in pageTags, that also exists in our whitelist
+    if (!pageTags) {
+        // If there are no page tags we will never find a match so
+        return false;
+    }
+
+    const isInWhitelist = (tag: string) => WHITELISTED_TAGS.includes(tag);
+
+    // For paid content we just return the first tag, otherwise we
+    // filter for the first tag in the whitelist
+    return isPaidContent ? pageTags[0] : pageTags.find(isInWhitelist);
```

Comment:
Is this ok to return `undefined`?

Reply:
I always thought `find` returned `false` when there's not match but you're right, it's `undefined`. Fixed